lib/chef/resource/aws_cache_subnet_group.rb

```diff
@@ -0,0 +1,22 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+class Chef::Resource::AwsCacheSubnetGroup < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type AWS::ElastiCache, id: :group_name
```

Comment:
Did you find a reason you needed to set the `id` field here?  I think its only used for the option lookup, which it doesn't seem like we need for these resources.